### ISSUE TITLE
Empty button false positives on WP core image lightbox controla

### ISSUE THREAD

**Issue by amberhinds** [equalizedigital/accessibility-checker#1883]
The following code snippets flag as empty buttons on https://shinny.wpenginepowered.com/resources/implementation-guides/

Close button:
```
<button type="button" style="fill:#000" class="wp-lightbox-close-button" data-wp-bind--aria-label="state.closeButtonAriaLabel">
<span class="wp-lightbox-close-icon" data-wp-bind--hidden="!state.hasNavigationIcon" hidden=""><svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24" width="20" height="20" aria-hidden="true" focusable="false"><path d="m13.06 12 6.47-6.47-1.06-1.06L12 10.94 5.53 4.47 4.47 5.53 10.94 12l-6.47 6.47 1.06 1.06L12 13.06l6.47 6.47 1.06-1.06L13.06 12Z"></path></svg></span>
<span class="wp-lightbox-close-text" data-wp-bind--hidden="!state.hasNavigationText" hidden="">Close</span>
</button>
```

<img width="1316" height="1002" alt="Image" src="https://github.com/user-attachments/assets/5cc34d46-4757-4d5f-a60d-470019f0b40d" />

Next button:
```
<button type="button" style="fill:#000" class="wp-lightbox-navigation-button wp-lightbox-navigation-button-next edac-highlight-element-selected" data-wp-bind--hidden="!state.hasNavigation" data-wp-on--click="actions.showNextImage" data-wp-bind--aria-label="state.nextButtonAriaLabel" hidden="">
					<span class="wp-lightbox-navigation-text" data-wp-bind--hidden="!state.hasNavigationText" hidden="">Next</span>
					<span class="wp-lightbox-navigation-icon" data-wp-bind--hidden="!state.hasNavigationIcon" hidden=""><svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24" width="28" height="28" aria-hidden="true" focusable="false"><path d="M10.6 6L9.4 7l4.6 5-4.6 5 1.2 1 5.4-6z"></path></svg></span>
				</button>
```

Previous button:
```
<button type="button" style="fill:#000" class="wp-lightbox-navigation-button wp-lightbox-navigation-button-prev" data-wp-bind--hidden="!state.hasNavigation" data-wp-on--click="actions.showPreviousImage" data-wp-bind--aria-label="state.prevButtonAriaLabel" hidden="">
					<span class="wp-lightbox-navigation-icon" data-wp-bind--hidden="!state.hasNavigationIcon" hidden=""><svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24" width="28" height="28" aria-hidden="true" focusable="false"><path d="M14.6 7l-1.2-1L8 12l5.4 6 1.2-1-4.6-5z"></path></svg></span>
					<span class="wp-lightbox-navigation-text" data-wp-bind--hidden="!state.hasNavigationText" hidden="">Previous</span>
				</button>
```

You can see in the code that a label is present but is hidden with a `hidden` attribute. When the lightbox is opened, `data-wp-bind--aria-label` adds the aria-label to the `button`. For example, here is the close button code after the lightboxed has been opened:
```
<button type="button" style="fill:#000" class="wp-lightbox-close-button" data-wp-bind--aria-label="state.closeButtonAriaLabel" aria-label="Close">
					<span class="wp-lightbox-close-icon" data-wp-bind--hidden="!state.hasNavigationIcon"><svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24" width="20" height="20" aria-hidden="true" focusable="false"><path d="m13.06 12 6.47-6.47-1.06-1.06L12 10.94 5.53 4.47 4.47 5.53 10.94 12l-6.47 6.47 1.06 1.06L12 13.06l6.47 6.47 1.06-1.06L13.06 12Z"></path></svg></span>
					<span class="wp-lightbox-close-text" data-wp-bind--hidden="!state.hasNavigationText" hidden="">Close</span>
				</button>
```
Note the addition of `aria-label="Close"`.

These buttons are caused by setting a WordPress core block to open in a lightbox. To replicate:
1. Insert a core image block.
2. Select an image.
3. Choose the option to "Enlarge on Click"
<img width="694" height="384" alt="Image" src="https://github.com/user-attachments/assets/e077bf72-64ec-4b6f-9f4d-c8574e4280eb" />

I understand why these are flagging because the scan happens with the image in it's unexpanded state and the aria-label doesn't exist then. It is an interesting example of a state-based false positive. 

Since it's in core, it could create a lot of noise. Do we have any idea about how we could address this?
